### ISSUE TITLE
RimWorld v1.5 Update

### ISSUE THREAD

**Issue by ubergarm** [ubergarm/RimWorld-KanbanStockpile#26]
Initial testing of compiling existing code throws this debug error on startup:
```
Error in static constructor of KanbanStockpile.KanbanStockpileLoader: System.TypeInitializationException: The type initializer for 'KanbanStockpile.KanbanStockpileLoader' threw an exception. ---> HarmonyLib.HarmonyException: Patching exception in method null ---> System.ArgumentException: Undefined target method for patch method static System.Void KanbanStockpile.Dialog_RenameZone_SetName_Patch::Prefix(Verse.Zone ___zone, System.String name)
[Ref 863C5FC]
 at HarmonyLib.PatchClassProcessor.PatchWithAttributes (System.Reflection.MethodBase& lastOriginal, System.Boolean unpatch) [0x00047] in <abec11463bc04855a5322a0a868aeb22>:0 
 at HarmonyLib.PatchClassProcessor.Patch () [0x0006e] in <abec11463bc04855a5322a0a868aeb22>:0 
   --- End of inner exception stack trace ---
[Ref 97DB6EE8]
 at HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) [0x00154] in <abec11463bc04855a5322a0a868aeb22>:0 
 at HarmonyLib.PatchClassProcessor.Patch () [0x000a0] in <abec11463bc04855a5322a0a868aeb22>:0 
 at HarmonyLib.Harmony.<PatchAll>b__10_0 (System.Type type) [0x00007] in <abec11463bc04855a5322a0a868aeb22>:0 
 at HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) [0x00014] in <abec11463bc04855a5322a0a868aeb22>:0 
 at HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) [0x00006] in <abec11463bc04855a5322a0a868aeb22>:0 
 at HarmonyLib.Harmony.PatchAll () [0x0001d] in <abec11463bc04855a5322a0a868aeb22>:0 
 at <0x40b3aa90 + 0x00043> <unknown method>
   --- End of inner exception stack trace ---
[Ref 7EE30E7E]
 at (wrapper managed-to-native) System.Runtime.CompilerServices.RuntimeHelpers.RunClassConstructor(intptr)
 at System.Runtime.CompilerServices.RuntimeHelpers.RunClassConstructor (System.RuntimeTypeHandle type) [0x0002a] in <eae584ce26bc40229c1b1aa476bfa589>:0 
 at Verse.StaticConstructorOnStartupUtility.CallAll () [0x00025] in <869ae09cd42e4958ad76d610931cb571>:0 
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
Verse.StaticConstructorOnStartupUtility:CallAll ()
Verse.PlayDataLoader/<>c:<DoPlayLoad>b__4_4 ()
Verse.LongEventHandler:ExecuteToExecuteWhenFinished ()
Verse.LongEventHandler:UpdateCurrentAsynchronousEvent ()
Verse.LongEventHandler:LongEventsUpdate (bool&)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition:Verse.Root.Update_Patch1 (Verse.Root)
Verse.Root_Entry:Update ()
```

Also saw this error when attempting to access a storage zone, might be related to above not actually loading quite right:
```
Exception filling tab RimWorld.ITab_Storage: System.MissingMethodException: void Verse.Widgets.CheckboxLabeled(UnityEngine.Rect,string,bool&,bool,UnityEngine.Texture2D,UnityEngine.Texture2D,bool)
[Ref C3186EE1]
 at RimWorld.ITab_Storage.FillTab () [0x00057] in <869ae09cd42e4958ad76d610931cb571>:0 
     - TRANSPILER net.ubergarm.rimworld.mods.kanbanstockpile: IEnumerable`1 KanbanStockpile.ITab_Storage_FillTab_Patch:Transpiler(IEnumerable`1 instructions)
 at Verse.InspectTabBase+<>c__DisplayClass19_0.<DoTabGUI>b__0 () [0x00039] in <869ae09cd42e4958ad76d610931cb571>:0 
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
Verse.Log:ErrorOnce (string,int)
Verse.InspectTabBase/<>c__DisplayClass19_0:<DoTabGUI>b__0 ()
Verse.ImmediateWindow:DoWindowContents (UnityEngine.Rect)
Verse.Window:InnerWindowOnGUI (int)
UnityEngine.GUI:CallWindowDelegate (UnityEngine.GUI/WindowFunction,int,int,UnityEngine.GUISkin,int,single,single,UnityEngine.GUIStyle)
```

**Comment by ubergarm** [comment]
Well, seems like the stockpile zone rename code in RimWorld v1.5 went to using generics which are trickier to patch. For now Just do the minimum update to get this working as much as possible in 1.5.

So unfortunately, this breaks preserving Kanban Stockpile settings when you `rename` a stockpile zone. Just be aware and manually set it back again if you rename your stockpile zone.

Might figure out how to target that again with Harmony...

**Comment by ubergarm** [comment]
The good news is that with just a few tweaks based on font size changes (or something?) it seems to be mostly working about the same!

**Comment by ubergarm** [comment]
Closed with mod version v1.1.2
